server/index.js

```diff
@@ -25,30 +25,137 @@ app.get('/api/products', (req, res, next) => {
                            "shortDescription"
-                    from "products"`;
+                            from "products"`;
 
   db.query(sqlQuery)
     .then(result => res.json(result.rows))
     .catch(err => next(err));
 });
 
 app.get('/api/products/:productId', (req, res, next) => {
+  const { productId } = req.params;
+  if (!productId || isNaN(parseInt(productId)) || productId < 0) {
+    next(new ClientError(`Expected an integer. ${productId} is not a invalid id.`, 400));
+    return;
+  }
+
   const sqlQuery = `select *
                     from "products"
                     where "productId" = $1`;
-  const value = [req.params.productId];
+  const value = [productId];
 
   db.query(sqlQuery, value)
     .then(result => {
       if (!result.rowCount) {
-        next(new ClientError(`There are no records matching id: ${req.params.productId}`, 404));
+        next(new ClientError(`There are no records matching id: ${productId}`, 404));
         return;
       }
       res.json(result.rows[0]);
     })
     .catch(err => next(err));
 });
 
+app.get('/api/cart', (req, res, next) => {
+  if (!req.session.cartId || req.session.cartId === null) return res.status(200).json([]);
+  const sqlSelectQuery = `select "c"."cartItemId",
+                              "c"."price",
+                              "p"."productId",
+                              "p"."image",
+                              "p"."name",
+                              "p"."shortDescription"
+                          from "cartItems" as "c"
+                          join "products" as "p" using ("productId")
+                          where "c"."cartId" = $1`;
+  const selectValues = [req.session.cartId];
+  db.query(sqlSelectQuery, selectValues)
+    .then(cartItems => {
+      return res.status(200).json(cartItems.rows);
+    })
+    .catch(err => next(err));
+});
+
+app.post('/api/cart', (req, res, next) => {
+  const productId = parseInt(req.body.productId);
+
+  if (!productId || isNaN(productId) || productId < 0) {
+    next(new ClientError(`Expected a positive integer. ${productId} is not a invalid id.`, 400));
+    return;
+  }
+
+  const sqlSelectQuery = `select "price"
+                          from "products"
+                          where "productId" = $1`;
+  const selectValues = [productId];
+  db.query(sqlSelectQuery, selectValues)
+    .then(selectResult => {
+      let sqlInsertQuery = `insert into "carts" ("cartId", "createdAt")
+                            values (default, default)
+                            returning "cartId"`;
+      let insertValues = [];
+      if (req.session.cartId && req.session.cartId !== null) {
+        sqlInsertQuery = `select "cartId"
+                          from "carts"
+                          where "cartId" = $1`;
+        insertValues = [req.session.cartId];
+      }
+      if (!selectResult.rowCount) {
+        next(new ClientError(`There is no price listed for the item with id: ${productId}`, 400));
```

Comment:
Be sure to `throw new ClientError(message, statusCode)` here instead of calling `next()`. Since we're in a promise chain, the rest of the code will still run unless an error is thrown or a rejected promise is returned.